### PR TITLE
meta: Notify Zulip for rustdoc nominated issues

### PR DESCRIPTION
**Pull request by camelid**
The rustdoc team does not currently use the `I-nominated` label, unlike
the libs and compiler teams (and maybe others). One reason for this is
that the other teams discuss their nominated issues in meetings, while
rustdoc is an async-only team.

However, it might be helpful to start using the `I-nominated` label for
rustdoc. The team currently uses a `cc @rust-lang/rustdoc` ping as the
equivalent, but it's easier to track issues when they use `I-nominated`.
Also we'd be more consistent with the other teams' procedures.

Since rustdoc doesn't have meetings, I propose we instead use the
triagebot notify Zulip functionality to create a topic in `#rustdoc` on
Zulip and ping the team. So it would look a bit like the procedure for
WG-prioritization when an issue acquires the `I-prioritize` label.

cc @rust-lang/rustdoc for approval of the change
r? @Mark-Simulacrum to make sure I configured triagebot correctly

### CODE REVIEW

File: triagebot.toml
Comment (camelid) [inline triagebot.toml:106]: I think I might need to swap `I-nominated` and `T-rustdoc` here so that other teams could have an `I-nominated` notification in the future.

```suggestion
[notify-zulip."T-rustdoc"]
required_labels = ["I-nominated"]
```

File: triagebot.toml
Comment (camelid) [inline triagebot.toml:108]: Another option: `nominated: #{number}`. I skipped putting the title in the topic because it gets confusing when the issue is renamed (I've experienced this before with WG-prioritization).

File: triagebot.toml
Comment (camelid) [inline triagebot.toml:110]:
```suggestion
@*T-rustdoc* issue #{number} "{title}" has been nominated for `T-rustdoc` discussion.
```

File: triagebot.toml
Comment (Manishearth) [reply triagebot.toml:108]: I like `nominated: #{number}`

File: triagebot.toml
Comment (Manishearth) [reply triagebot.toml:106]: +1

**Review by Manishearth — APPROVED** [review]

**Comment by jyn514** [comment]
This was a good idea :)

File: triagebot.toml
Comment (camelid) [reply triagebot.toml:108]:
```suggestion
topic = "nominated: #{number}"
```

**Comment by Manishearth** [comment]
@camelid feel free to r=me once you've figured out what to do about your comments on the PR.

**Comment by camelid** [comment]
(Squashed.)

File: triagebot.toml
Comment (camelid) [inline triagebot.toml:112]: The one issue with this setup is that we will only be notified if *`T-rustdoc`* is removed—nothing will happen if `I-nominated` is removed. It might be worth getting rid of the `message_on_remove` since it will be kind of confusing.

**Comment by camelid** [comment]
I think I'll wait until one or two more people approve, and for feedback on https://github.com/rust-lang/rust/pull/82689#discussion_r585196415, before I merge.

File: triagebot.toml
Comment (Manishearth) [reply triagebot.toml:112]: Does an array work here? cc @Mark-Simulacrum can we do `[[notify-zulip."I-nominated"]]` here?

Comment (Mark-Simulacrum) [reply triagebot.toml:112]: No, I don't think so, but I also don't really know what you mean by that. That said, it seems quite reasonable for triagebot to detect the removal of a required label and treat that just like removal of T-rustdoc; I think the reasonable treatment is that the whole set of labels must be present and removal of any leads to a notification of such. Happy to accept a PR to that effect.

**Comment by Mark-Simulacrum** [comment]
Broadly looks good to me, but these things are finicky, so I recommend merging and trying it out.

File: triagebot.toml
Comment (camelid) [reply triagebot.toml:112]: > That said, it seems quite reasonable for triagebot to detect the removal of a required label and treat that just like removal of T-rustdoc; I think the reasonable treatment is that the whole set of labels must be present and removal of any leads to a notification of such. Happy to accept a PR to that effect.

That might be a natural extension of the work in rust-lang/triagebot#1078.

Comment (camelid) [reply triagebot.toml:112]: > No, I don't think so, but I also don't really know what you mean by that.

I think he means that we would be notified as soon as both `I-nominated` and `T-rustdoc` were present, and we would be notified again as soon as one of them was removed. Treating the removal of a required label the same as the removal of the main label would have the same effect, but gives the labels an asymmetric relationship (`T-rustdoc` is considered the "main" label by triagebot).

**Review by jyn514 — APPROVED** [review]

**Review by GuillaumeGomez — APPROVED** [review]

**Pull request merged**
